ACPs/108-evm-event-importing/README.md

```diff
@@ -0,0 +1,117 @@
+
+With a standard interface and abstract contract implemetation that handles the authentication of block hashes and verification of Merkle proofs, smart contract developers on Avalanche will be able to much more easily create applications that leverage data from other Avalanche blockchains. These type of cross-chain application do not require any direct interaction on the source chain.
+
+## Specification
+
+### Event Importing Interface
+
+We propose that smart contracts importing EVM events emitted by other blockchains within Avalanche implement the following interface.
+
+#### Methods
+
+Imports the EVM event uniquely identified by the source blockchain ID, block header, transaction index, and log index. The event data is provided in the `receiptProof`, which must be a valid Merkle proof of the receipt for the given `txIndex` against the receipt root of the `blockHeader`. Must emit an `EventImported` event upon success. Should validate that the `blockHeader` matches an authenticated block hash from the `sourceBlockchainID`.
+```solidity
+function importEvent(
+    bytes32 sourceBlockchainID,
+    bytes calldata blockHeader,
+    uint256 txIndex,
+    bytes[] calldata receiptProof,
+    uint256 logIndex
+) external;
+```
+
+This interface does not require that the Warp precompile is used to authenticate block hashes. Implementations could:
+- Use an external block hash verification mechanism, such as trusted oracles.
+- Use the Warp precompile to authenticate block hashes provided in the transaction calling `importEvent`.
+- Look up previously authenticated block hashes from another contract, such that multiple events could be imported from the same block hash, which would only need to be authenticated once.
+
+#### Events
+
+Must trigger when an EVM event is imported.
+```solidity
+event EventImported(
+    bytes32 indexed sourceBlockchainID,
+    bytes32 indexed sourceBlockHash,
+    address indexed loggerAddress,
+    uint256 txIndex,
+    uint256 logIndex
+);
+```
+
+### Event Importing Abstract Contract
+
+Applications importing EVM events emitted by other blockchains within Avalanche should be able to use a standard abstract implementation of the `importEvent` interface. This abstract implementation should handle:
```

Comment:
nit: IMO it is weird to use `should handle: authenticating block hashes using warp precompile` given you say `does not require the warp precompile` earlier?

That to me is a `could` with the others being `must`?

Reply:
Thanks for calling this out. I tweaked the interface to include the explicit `sourceBlockchainID` param after typing the initial draft, and agree this part was left unclear/inconsistent. Updated the wording now to `must`, but removed the "using the Warp precompile" qualifier and added an explanation of how implementations `could` authenticate the block hashes below.